eip-6170.md

```diff
@@ -0,0 +1,79 @@
+
+The lack of standardization not only makes integration difficult but also is error prone as developers need to understand the operation of the messaging protocol at its core.
+
+A standard for cross-chain messaging will lower integration efforts for cross-chain messaging while creating more robust & secure cross-chain application development.
+
+To the worst, each messaging protocol has its chain ids (or) chain identification mechanism. However, this mechanism is used as they support non-EVM chains, which makes integration a nightmare.
+
+## Specification
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+### Methods
+
+sendMessage
+Dispatch message/state to `receiver` on a predetermined `chain id`.
+MUST emit `MessageSent` event.
+MUST support global chain-ids. Propreitory chain ids should be mapped internally to global chain ids.
+MUST revert if `chain id` is invalid.
+MUST return `true` on successful response from underlying messaging protocol.
+
+**NOTE:** I'm proposing chain-ids be the bytes encoding of their native token name string. For eg., abi.encode("ETH"), abi.encode("SOL") imagining they cannot override.
+       
+       function sendMessage(bytes chainId, bytes receiver, bytes message) public returns(bool)
+
+receiveMessage
+Receive message/state from `sender` on a predetermined `chain id`
+MUST emit `MessageReceived` event.
+MUST support global based chain-ids.
+MUST revert if `sender` or `chain-id` is invalid.
+MUST return `true` on successful processing.
+
+**NOTE:** Sender validation (or) message validation, gas overrides are handled at relayer level.
+
+       function receiveMessage(bytes chainId, bytes sender, bytes message) public returns(bool)
+
+### Events
+
+MessageSent
+MUST trigger when message is sent, include zero bytes transfers.
+       
+       event MessageSent(bytes _sender, bytes _fromChainId, bytes _receiver, bytes _toChainId, bytes message)
+
+MessageReceived
+MUST trigger on any successful call to `receiveMessage(bytes chainId, bytes sender, bytes message)` function. 
+       
+       event MessageReceived(bytes _sender, bytes _fromChainId, bytes _receiver, bytes _toChainId, bytes message)
+
+## Rationale
+
+The Cross-Chain interface is designed to be optimized for messaging layer integrators with a feature complete yet minimal interface. Validations such as sender authentication, receiver whitelisting, relayer mechanisms and cross-chain execution overrides are intentionally not specified, as Messaging protocols are expected to be treated as black boxes on-chain and inspected off-chain before use.
+
+## Security Considerations
+
+Fully permissionless messaging could be a security threat to the protocol. It is recommended that all the integrators review the implementation of messaging tunnels before integrating.
```

Comment:
> Fully permissio'nless messaging could be a security threat to the protocol.

How so?

Reply:
For eg., without sender authentication, anyone could write arbitrary messages into the receiving smart contract. 

This EIP focuses only on the way the messages should be sent and received with a specific standard. But any authentication (or) message tunnel specific operations should be made inside the receive function and should be handled by the integrators or through wrapper contracts.

Reply:
Please put that in the EIP.